File: trees/java/MainArbol.java

public class MainArbol {

  public static void main(String[] args) {

    Arbol objArbol = new Arbol(
        "a",
        new Arbol(
          "b",
          new Arbol("d", null, null),
          new Arbol("e", null, null)
          ),
        new Arbol(
          "c",
          new Arbol(
            "f", null, null
          ),
          new Arbol(
            "g",
            new Arbol("k", null, null),
            null
          )
        )
    );

    System.out.println(objArbol);
    System.out.println(objArbol.getKey());
    System.out.println(objArbol.getLeft().getKey());
    System.out.println(objArbol.getRight().getKey());
  }
}

File: trees/typescript/MainArbol.ts
import { Arbol } from "./Arbol";


const objArbol = new Arbol<string>(
  "a",
  new Arbol<string>(
    "b",
    new Arbol<string>("d", null, null),
    new Arbol<string>("e", null, null)
  ),
  new Arbol<string>(
    "c",
    new Arbol<string>("f", null, null),
    new Arbol<string>(
      "g",
      new Arbol<string>("k", null, null),
      null
    )
  )
)

console.log(objArbol)
console.log(objArbol.key)
console.log(objArbol.left)
console.log(objArbol.right)
